src/components/notification-picker.tsx

```diff
@@ -0,0 +1,226 @@
+        });
+      }
+    },
+    [selectedEvents, model, modelChange]
+  );
+
+  const sendToChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    setSendToInput(e.target.value);
+  }, []);
+
+  const blur = useCallback(() => {
+    const emailArray = sendToInput
+      .split(',')
+      .map(email => email.trim())
+      .filter(email => email);
+    const updatedNotification = { ...model.notification, sendTo: emailArray };
+    modelChange({ ...model, notification: updatedNotification });
+  }, [sendToInput, model, modelChange]);
+
+  const keyDown = useCallback(
+    (e: React.KeyboardEvent) => {
+      if (e.key === 'Enter') {
+        e.preventDefault();
+        blur();
+      }
+    },
+    [blur]
+  );
+
+  const includeOutputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const updatedValue = event.target.checked;
+    setIncludeOutput(updatedValue);
+    modelChange({
+      ...model,
+      notification: {
+        ...model.notification,
+        includeOutput: updatedValue
+      }
+    });
+  };
+
+  const deleteSelectedEvent = useCallback(
+    (eventToDelete: string) => () => {
+      const updatedEvents = selectedEvents.filter(
+        event => event !== eventToDelete
+      );
+      setSelectedEvents(updatedEvents);
+      modelChange({
+        ...model,
+        notifications: { ...model.notification, selectedEvents: updatedEvents }
+      });
+    },
+    [selectedEvents, model, modelChange]
+  );
+
+  if (!notificationEvents.length) {
+    return null;
+  }
+
+  return (
+    <Stack size={2}>
+      <InputLabel>{trans.__('Notification')}</InputLabel>
+      <FormControlLabel
+        control={
+          <Switch
+            checked={enableNotification}
+            onChange={enableNotificationChange}
+          />
+        }
+        label={trans.__('Enable Notification')}
+      />
+      <TextField
+        label={trans.__('Send To')}
+        value={sendToInput}
+        name="sendTo"
+        variant="outlined"
+        onChange={sendToChange}
+        onBlur={blur}
+        onKeyDown={keyDown}
+        disabled={!enableNotification}
+      />
+      <NotificationEventsSelect
+        id={id}
+        availableEvents={notificationEvents.filter(
+          e => !selectedEvents.includes(e)
+        )}
+        selectChange={selectChange}
+        disabled={!enableNotification}
+      />
+      <SelectedEventsChips
+        selectedEvents={selectedEvents}
+        deleteSelectedEvent={deleteSelectedEvent}
+        disabled={!enableNotification}
+      />
+      <FormControlLabel
+        control={
+          <Checkbox
+            checked={includeOutput}
+            onChange={includeOutputChange}
+            disabled={!enableNotification}
+          />
+        }
+        label={trans.__('Include Output')}
+      />
+    </Stack>
+  );
+}
+
+const NotificationEventsSelect: React.FC<{
+  id: string;
+  availableEvents: string[];
+  selectChange: (e: SelectChangeEvent<string>) => void;
```

Comment:
Also, for input components, I'd prefer us to stick to the convention of always labeling the input state & change handler as `value` and `onChange` respectively.